### DISCUSSION TITLE
nextjs api works fine locally but in vercel I have the error `Error: Cannot find module '@prisma/client'`

### DISCUSSION THREAD

**Discussion by ivanlen** [prisma/prisma#23287] [Q&A]
### Question

Hi, I am using prisma, vercel, and nextjs in my project.

Everything is working fine locally (I can use the api without problems) but when I deploy to vercel and run from there I have the error
```
Error: Cannot find module '@prisma/client' Require stack:...
```

- in `pages/api/update_coords/index.js` I have

```
import prisma from "lib/prisma";

....
```

- in lib/prisma.ts I have
```
import { PrismaClient } from '@prisma/client';

let prisma: PrismaClient;

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;

```

What I find weird is that I am importing prisma the same way (import prisma from "lib/prisma";) in another file and I call a prisma from

- lib/queries.js

```
import prisma from "lib/prisma";

...

async function getAllTokens(contract = contractName) {
  const tokens = await prisma.tokens.findMany({
    where: { contract: { equals: contract } },
  });
  return tokens;
}

...
```
- and then I call from pages/index.js

```
export async function getServerSideProps() {
  const data = await getAllTokens()

  return {
    props: { data }
  }
}
```

and this works!!!!!

So prisma client is working when I call it from getServerSideProps but it is not when I use it in api/

And in package.json I have the following lines
```
"scripts": {
    "dev": "next dev",
    "build": "prisma generate && next build",
    "build:dev": "set NODE_ENV=developmnet & next build",
    "vercer-build": "prisma generate && next build",
    "start": "next start",
    "test": "NODE_ENV=test mocha --require @babel/register --require ignore-styles",
    "run": "--require @babel/register --require ignore-styles",
    "postinstall": "prisma generate"
  },
```

Its been 2 days of googling and tried lots of solutions and hacks, but I can't make this work.
Any idea?

some relevant packages

@babel/core@7.21.0
├── @chakra-ui/icons@2.0.10
├── @chakra-ui/react@2.4.9
├── @prisma/client@5.10.2
├── @vercel/og@0.6.2
├── next@13.4.6
├── prisma@5.10.2
├── react-dom@18.2.0
├── react-map-gl@7.0.18
├── react-router-dom@6.3.0
├── react@18.2.02
├── vercel@33.5.2

**Comment by ludralph** [comment]
Hi @ivanlen 👋 

Thank you for raising this question.

According to your package.json, you have a custom script `"vercer-build": "prisma generate && next build"`.  This looks like a typo and you meant to write `"vercel-build"`. Can you make the change and let us know if it fixes it for you?

**Reply by ivanlen** [reply]
I am can't be 100% it was this, but it was definitely a problem with the build, because after using `VERCEL_FORCE_NO_BUILD_CACHE=1` and forcing to rebuild from github things solved.
Maybe it wasn't solving before because of the typo in you mentioned.

You have a really good eye, nice catch!!

Also I removed git cached files [following this](https://stackoverflow.com/questions/62378045/how-to-fix-next-js-vercel-deployment-module-not-found-error).
